rules/DeadCode/SideEffect/SideEffectNodeDetector.php

```diff
@@ -43,41 +30,10 @@ final class SideEffectNodeDetector
     ];
 
     public function __construct(
-        private readonly NodeTypeResolver $nodeTypeResolver,
         private readonly PureFunctionDetector $pureFunctionDetector
     ) {
     }
 
-    public function detect(Expr $expr, Scope $scope): bool
```

Comment:
this `detect` method seems only used by `TernaryFalseExpressionToIfRector`, which there is already `detectCallExpr()` for it, so I removed it and just use `detectCallExpr()`.

Reply:
I added more fixture for assign in else, seems some logic need to be merged, eg: `$expr instanceof Assign`

Reply:
I changed `detectCallExpr()` to `detect()` with `Assign` check.